examples/custom-rlpx-subprotocol/src/subprotocol/protocol/message.rs

```diff
@@ -0,0 +1,88 @@
+use reth_eth_wire::{protocol::Protocol, Capability};
+use reth_primitives::{Buf, BufMut, BytesMut};
+
+#[repr(u8)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub(crate) enum CustomRlpxProtoMessageId {
+    Ping = 0x00,
+    Pong = 0x01,
+    CustomMessage = 0x02,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) enum CustomRlpxProtoMessageKind {
+    Ping,
+    Pong,
+    CustomMessage(String),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct CustomRlpxProtoMessage {
+    pub message_type: CustomRlpxProtoMessageId,
+    pub message: CustomRlpxProtoMessageKind,
+}
+
+impl CustomRlpxProtoMessage {
+    /// Returns the capability for the `custom_rlpx` protocol.
+    pub fn capability() -> Capability {
+        Capability::new_static("custom_rlpx", 1)
+    }
+
+    /// Returns the protocol for the `custom_rlpx` protocol.
+    pub fn protocol() -> Protocol {
+        Protocol::new(Self::capability(), 3)
+    }
+
+    /// Creates a ping message
+    pub fn ping() -> Self {
+        Self {
+            message_type: CustomRlpxProtoMessageId::Ping,
+            message: CustomRlpxProtoMessageKind::Ping,
+        }
+    }
+
+    /// Creates a pong message
+    pub fn pong() -> Self {
+        Self {
+            message_type: CustomRlpxProtoMessageId::Pong,
+            message: CustomRlpxProtoMessageKind::Pong,
+        }
+    }
+
+    /// Creates a new `CustomRlpxProtoMessage` with the given message ID and payload.
+    pub fn encoded(&self) -> BytesMut {
+        let mut buf = BytesMut::new();
```

Comment:
```suggestion
        let mut buf = Vec::new();
```

Reply:
@mattsse i cant change this as the ConnectionHandler expects the associated type as a Stream of BytesMut:

https://github.com/paradigmxyz/reth/blob/74f5a852365c451217cbfa49d90bb55623c05b23/crates/net/network/src/protocol.rs#L47